src/io/OhmmsData/XMLParsingString.h

```diff
@@ -19,10 +19,49 @@
 #define QMCPLUSPLUS_XMLSTRING_H
 
 #include <string>
+#include <string_view>
 #include <libxml/xmlmemory.h>
 #include <libxml/tree.h>
 
+// I think these should be in libxmldefs.h
+// but it includes XMLParsingString so these cast functions need to be here.
+
+/** xmlChar* to char* cast
+ *  certainly not typesafe but at this interface between libxml and c++
+ *  well it beats c style casts and might be more correct than a reinterpret.
+ *
+ *  This is fine with UTF-8 bytes going into a std::string.
+ */
+inline char* castXMLCharToChar(xmlChar* c)
+{
+  return static_cast<char*>(static_cast<void *>(c));
```

Comment:
What is the reason for `void *`?

Reply:
Apparently guarantees that the address doesn't change. But it seems a bit controversial, some claim since c++11 reinterpret_cast means the same thing.  I think maybe since all the types involved here are of the same size on most platforms it never mattered. 
I think in c++17 it can be replaced with a reinterpret_cast at no loss but that looks like an emergency.